### PR TITLE
Decouple PipelineDDLGenerator with ContextManager for common usage

### LINKED ISSUE #20141: Decouple PipelineDDLGenerator with ContextManager for common usage
Part of #19421.

Currently, PipelineDDLGenerator depends on ContextManager, ShardingSphereDatabase and Map<String,DataSource> will be got from ContextManager.
When importing data from unmanaged resource (database), we could not get DataSource from ContextManager.

Possible changes:
- Update parameters of generateLogicDDLSQL method to: databaseType, dataSource, schemaName, actualTableName, logicTableName, ShardingSphereSQLParserEngine
- Remove dependency of ContextManager from PipelineDDLGenerator

### PR DESCRIPTION
**Pull request by huangdx0726**
Fixes #20141 .

Changes proposed in this pull request:
- Update parameters of generateLogicDDLSQL method to: databaseType, dataSource, schemaName, actualTableName, logicTableName, ShardingSphereSQLParserEngine
- Remove dependency of ContextManager from PipelineDDLGenerator
-

### CODE REVIEW

**Comment by sandynz** [comment]
@huangdx0726 , scaling-it-test ran failed, the job status is `PREPARING_FAILURE`. Could you do some debug on local?

Related exception:
```
Error:  2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: [ERROR] 2022-08-15 16:43:44.094 [Thread-20] o.a.s.d.p.s.migration.MigrationJob - job prepare failed, j0130317c3054317c7368617264696e675f6462-0
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: org.apache.shardingsphere.data.pipeline.core.exception.PipelineJobPrepareFailedException: prepare target tables failed.
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.mysql.prepare.datasource.MySQLDataSourcePreparer.prepareTargetTables(MySQLDataSourcePreparer.java:55)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.core.prepare.PipelineJobPreparerUtils.prepareTargetTables(PipelineJobPreparerUtils.java:81)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.scenario.migration.MigrationJobPreparer.prepareTarget(MigrationJobPreparer.java:139)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.scenario.migration.MigrationJobPreparer.prepareAndCheckTarget(MigrationJobPreparer.java:117)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.scenario.migration.MigrationJobPreparer.prepareAndCheckTargetWithLock(MigrationJobPreparer.java:103)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.scenario.migration.MigrationJobPreparer.prepare(MigrationJobPreparer.java:68)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.scenario.migration.MigrationJob.prepare(MigrationJob.java:79)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.scenario.migration.MigrationJob.lambda$execute$0(MigrationJob.java:70)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at java.lang.Thread.run(Thread.java:748)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Table 'scaling_it_4.t_order_0' doesn't exist
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at com.mysql.jdbc.Util.getInstance(Util.java:408)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:944)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3978)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3914)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2530)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2683)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionIm
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: pl.java:2491)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2449)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at com.mysql.jdbc.StatementImpl.executeQuery(StatementImpl.java:1381)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at com.zaxxer.hikari.pool.ProxyStatement.executeQuery(ProxyStatement.java:111)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at com.zaxxer.hikari.pool.HikariProxyStatement.executeQuery(HikariProxyStatement.java)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.driver.executor.callback.impl.StatementExecuteQueryCallback.executeQuery(StatementExecuteQueryCallback.java:40)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.driver.executor.callback.ExecuteQueryCallback.executeSQL(ExecuteQueryCallback.java:45)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.driver.executor.callback.ExecuteQueryCallback.executeSQL(ExecuteQueryCallback.java:37)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.infra.executor.sql.execute.engine.driver.jdbc.JDBCExecutorCallback.execute(JDBCExecutorCallback.java:95)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.infra.executor.sql.execute.engine.driver.jdbc.JDBCExecutorCallback.execute(JDBCExecutorCallback.java:75)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.infra.executor.kernel.ExecutorEngine.syncExecute(ExecutorEngine.java:135)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.infra.executor.kernel.ExecutorEngine.parallelExecute(ExecutorEngine.java:131)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.infra.executor.kernel.ExecutorEngine.execute(ExecutorEngine.java:116)
[INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.infra.executor.sql.execute.engine.driver.jdbc.JDBCExecutor.execute(JDBCExecutor.java:65)
[INFO ] 2022-08-15 16:43:44.310 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.infra.executor.sql.execute.engine.driver.jdbc.JDBCExecutor.execute(JDBCExecutor.java:49)
[INFO ] 2022-08-15 16:43:44.310 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.driver.executor.DriverJDBCExecutor.executeQuery(DriverJDBCExecutor.java:82)
[INFO ] 2022-08-15 16:43:44.310 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.driver.jdbc.core.statement.ShardingSphereStatement.executeQuery0(ShardingSphereStatement.java:218)
[INFO ] 2022-08-15 16:43:44.310 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.driver.jdbc.core.statement.ShardingSphereStatement.executeQuery(ShardingSphereStatement.java:169)
[INFO ] 2022-08-15 16:43:44.310 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.mysql.ddlgenerator.MySQLCreateTableSQLGenerator.generate(MySQLCreateTableSQLGenerator.java:45)
[INFO ] 2022-08-15 16:43:44.310 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.core.metadata.generator.PipelineDDLGenerator.generateActualDDLSQL(PipelineDDLGenerator.java:139)
[INFO ] 2022-08-15 16:43:44.310 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.core.metadata.generator.PipelineDDLGenerator.generateLogicDDLSQL(PipelineDDLGenerator.java:89)
[INFO ] 2022-08-15 16:43:44.310 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.mysql.prepare.datasource.MySQLDataSourcePreparer.getCreateTableSQL(MySQLDataSourcePreparer.java:71)
[INFO ] 2022-08-15 16:43:44.310 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.mysql.prepare.datasource.MySQLDataSourcePreparer.prepareTargetTables(MySQLDataSourcePreparer.java:50)
[INFO ] 2022-08-15 16:43:44.310 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	... 8 common frames omitted
```

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-mysql/src/main/java/org/apache/shardingsphere/data/pipeline/mysql/prepare/datasource/MySQLDataSourcePreparer.java [hunk 1]
```diff
@@ -52,11 +57,19 @@ public void prepareTargetTables(final PrepareTargetTablesParameter parameter) {
     }
     
     private List<String> getCreateTableSQL(final PrepareTargetTablesParameter parameter) {
-        PipelineDDLGenerator generator = new PipelineDDLGenerator(PipelineContext.getContextManager());
+        PipelineDDLGenerator generator = new PipelineDDLGenerator();
+        DataSource dataSource = parameter.getDataSourceManager().getDataSource(parameter.getDataSourceConfig());
+        ShardingSphereMetaData metaData = PipelineContext.getContextManager().getMetaDataContexts().getMetaData();
```
Comment (sandynz) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-mysql/src/main/java/org/apache/shardingsphere/data/pipeline/mysql/prepare/datasource/MySQLDataSourcePreparer.java:62]: `DataSource dataSource = parameter.getDataSourceManager().getDataSource(parameter.getDataSourceConfig());` might return ShardingSphereDataSource, could we reference the original code `database.getResource().getDataSources().get(dataSourceName)` in PipelineDDLGenerator?

And also PostgreSQL and openGauss implementations.

You could refer to:
- PrepareTargetTablesParameter.tablesFirstDataNodes
- JobDataNodeLine, JobDataNodeEntry (logicTableName, dataNodes)
- DataNode (dataSourceName, tableName), e.g. ds_0.t_order_0

**Review by sandynz — CHANGES_REQUESTED** [review]

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/core/metadata/generator/PipelineDDLGenerator.java [hunk 2]
```diff
@@ -67,27 +65,31 @@ public final class PipelineDDLGenerator {
...
+     * @param databases databases
+     * @param parserEngine parser engine
      * @return ddl SQL
      */
     @SneakyThrows
-    public String generateLogicDDLSQL(final DatabaseType databaseType, final String databaseName, final String schemaName, final String tableName) {
-        ShardingSphereDatabase database = contextManager.getMetaDataContexts().getMetaData().getDatabase(databaseName);
+    public String generateLogicDDLSQL(final DataSource dataSource, final String databaseName, final String schemaName,
+                                      final String logicTableName, final String actualTableName,
+                                      final Map<String, ShardingSphereDatabase> databases, final ShardingSphereSQLParserEngine parserEngine) {
```
[the remark this answers is not in the document]
Comment (huangdx0726) [reply shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/core/metadata/generator/PipelineDDLGenerator.java:84]: I’ll fix it

**Comment by huangdx0726** [comment]
> @huangdx0726 , scaling-it-test ran failed, the job status is `PREPARING_FAILURE`. Could you do some debug on local?
> 
> Related exception:
> 
> ```
> Error:  2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: [ERROR] 2022-08-15 16:43:44.094 [Thread-20] o.a.s.d.p.s.migration.MigrationJob - job prepare failed, j0130317c3054317c7368617264696e675f6462-0
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: org.apache.shardingsphere.data.pipeline.core.exception.PipelineJobPrepareFailedException: prepare target tables failed.
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.mysql.prepare.datasource.MySQLDataSourcePreparer.prepareTargetTables(MySQLDataSourcePreparer.java:55)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.core.prepare.PipelineJobPreparerUtils.prepareTargetTables(PipelineJobPreparerUtils.java:81)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.scenario.migration.MigrationJobPreparer.prepareTarget(MigrationJobPreparer.java:139)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.scenario.migration.MigrationJobPreparer.prepareAndCheckTarget(MigrationJobPreparer.java:117)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.scenario.migration.MigrationJobPreparer.prepareAndCheckTargetWithLock(MigrationJobPreparer.java:103)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.scenario.migration.MigrationJobPreparer.prepare(MigrationJobPreparer.java:68)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.scenario.migration.MigrationJob.prepare(MigrationJob.java:79)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.scenario.migration.MigrationJob.lambda$execute$0(MigrationJob.java:70)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at java.lang.Thread.run(Thread.java:748)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Table 'scaling_it_4.t_order_0' doesn't exist
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at com.mysql.jdbc.Util.getInstance(Util.java:408)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:944)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3978)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3914)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2530)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2683)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionIm
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: pl.java:2491)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2449)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at com.mysql.jdbc.StatementImpl.executeQuery(StatementImpl.java:1381)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at com.zaxxer.hikari.pool.ProxyStatement.executeQuery(ProxyStatement.java:111)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at com.zaxxer.hikari.pool.HikariProxyStatement.executeQuery(HikariProxyStatement.java)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.driver.executor.callback.impl.StatementExecuteQueryCallback.executeQuery(StatementExecuteQueryCallback.java:40)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.driver.executor.callback.ExecuteQueryCallback.executeSQL(ExecuteQueryCallback.java:45)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.driver.executor.callback.ExecuteQueryCallback.executeSQL(ExecuteQueryCallback.java:37)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.infra.executor.sql.execute.engine.driver.jdbc.JDBCExecutorCallback.execute(JDBCExecutorCallback.java:95)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.infra.executor.sql.execute.engine.driver.jdbc.JDBCExecutorCallback.execute(JDBCExecutorCallback.java:75)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.infra.executor.kernel.ExecutorEngine.syncExecute(ExecutorEngine.java:135)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.infra.executor.kernel.ExecutorEngine.parallelExecute(ExecutorEngine.java:131)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.infra.executor.kernel.ExecutorEngine.execute(ExecutorEngine.java:116)
> [INFO ] 2022-08-15 16:43:44.309 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.infra.executor.sql.execute.engine.driver.jdbc.JDBCExecutor.execute(JDBCExecutor.java:65)
> [INFO ] 2022-08-15 16:43:44.310 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.infra.executor.sql.execute.engine.driver.jdbc.JDBCExecutor.execute(JDBCExecutor.java:49)
> [INFO ] 2022-08-15 16:43:44.310 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.driver.executor.DriverJDBCExecutor.executeQuery(DriverJDBCExecutor.java:82)
> [INFO ] 2022-08-15 16:43:44.310 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.driver.jdbc.core.statement.ShardingSphereStatement.executeQuery0(ShardingSphereStatement.java:218)
> [INFO ] 2022-08-15 16:43:44.310 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.driver.jdbc.core.statement.ShardingSphereStatement.executeQuery(ShardingSphereStatement.java:169)
> [INFO ] 2022-08-15 16:43:44.310 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.mysql.ddlgenerator.MySQLCreateTableSQLGenerator.generate(MySQLCreateTableSQLGenerator.java:45)
> [INFO ] 2022-08-15 16:43:44.310 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.core.metadata.generator.PipelineDDLGenerator.generateActualDDLSQL(PipelineDDLGenerator.java:139)
> [INFO ] 2022-08-15 16:43:44.310 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.core.metadata.generator.PipelineDDLGenerator.generateLogicDDLSQL(PipelineDDLGenerator.java:89)
> [INFO ] 2022-08-15 16:43:44.310 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.mysql.prepare.datasource.MySQLDataSourcePreparer.getCreateTableSQL(MySQLDataSourcePreparer.java:71)
> [INFO ] 2022-08-15 16:43:44.310 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	at org.apache.shardingsphere.data.pipeline.mysql.prepare.datasource.MySQLDataSourcePreparer.prepareTargetTables(MySQLDataSourcePreparer.java:50)
> [INFO ] 2022-08-15 16:43:44.310 [docker-java-stream--142075617] :ShardingSphere-Proxy - STDOUT: 	... 8 common frames omitted
> ```

ok，

**Review by sandynz — APPROVED** [review]
Need more review by @tuichenchuxin

**Review by tuichenchuxin — APPROVED** [review]

**Pull request merged**